Arrowgene.Ddon.Database/Migrations/00000001_DatabaseCreate.cs

```diff
@@ -0,0 +1,574 @@
+                .WithColumn("shortcut_id").AsInt32().NotNullable()
+                .WithColumn("u32_data").AsInt32().NotNullable()
+                .WithColumn("f32_data").AsInt32().NotNullable()
+                .WithColumn("exex_type").AsInt16().NotNullable();
+
+            Create.Table("ddon_communication_shortcut")
+                .WithColumn("character_id").AsInt32().NotNullable()
+                    .PrimaryKey()
+                    .ForeignKey("ddon_character", "character_id").OnDelete(System.Data.Rule.Cascade)
+                .WithColumn("page_no").AsInt32().NotNullable()
+                    .PrimaryKey()
+                .WithColumn("button_no").AsInt32().NotNullable()
+                    .PrimaryKey()
+                .WithColumn("type").AsInt16().NotNullable()
+                .WithColumn("category").AsInt16().NotNullable()
+                .WithColumn("id").AsInt32().NotNullable();
+
+            Create.Table("ddon_pawn_reaction")
+                .WithColumn("pawn_id").AsInt32().NotNullable()
+                    .PrimaryKey()
+                    .ForeignKey("ddon_pawn", "pawn_id").OnDelete(System.Data.Rule.Cascade)
+                .WithColumn("reaction_type").AsInt16().NotNullable()
+                    .PrimaryKey()
+                .WithColumn("motion_no").AsInt32().NotNullable();
+
+            Create.Table("ddon_sp_skill")
+                .WithColumn("pawn_id").AsInt32().NotNullable()
+                    .PrimaryKey()
+                    .ForeignKey("ddon_pawn", "pawn_id").OnDelete(System.Data.Rule.Cascade)
+                .WithColumn("job").AsInt16().NotNullable()
+                    .PrimaryKey()
+                .WithColumn("sp_skill_id").AsInt16().NotNullable()
+                    .PrimaryKey()
+                .WithColumn("sp_skill_lv").AsInt16().NotNullable();
+
+            Create.Table("ddon_released_warp_point")
+                .WithColumn("character_id").AsInt32().NotNullable()
+                    .PrimaryKey()
+                    .ForeignKey("ddon_character", "character_id").OnDelete(System.Data.Rule.Cascade)
+                .WithColumn("warp_point_id").AsInt32().NotNullable()
+                    .PrimaryKey()
+                .WithColumn("favorite_slot_no").AsInt32().NotNullable();
+
+            Create.Table("ddon_game_token")
+                .WithColumn("account_id").AsInt32().NotNullable()
+                    .PrimaryKey()
+                    .ForeignKey("account", "id").OnDelete(System.Data.Rule.Cascade)
+                .WithColumn("character_id").AsInt32().NotNullable()
+                    .ForeignKey("ddon_character", "character_id").OnDelete(System.Data.Rule.Cascade)
+                .WithColumn("token").AsString().NotNullable()
+                    .Unique()
+                .WithColumn("created").AsDateTime().NotNullable();
+
+            Create.Table("ddon_connection")
+                .WithColumn("server_id").AsInt32().NotNullable()
+                    // .Unique()
```

Comment:
I was having trouble to keep these two fields unique. It seems like maybe some logic is violated or I don't really understand the unique syntax. We try to insert a record before deleting the previous one for the same character. This causes a constraint violation to occur. When looking at the DB viewer tool, it seems like it doesn't consider the rules populated by the original schema to indicate these fields are actually unique.